ufl/form.py

```diff
@@ -49,12 +49,19 @@ def _sorted_integrals(integrals):
 
     all_integrals = []
 
+    def keyfunc(item):
+        if isinstance(item, int):
```

Comment:
This produces an error inside of Firedrake because sometimes `item` has type `numpy.int32`. This would go away if this were changed to [`numbers.Integral`](https://docs.python.org/3.8/library/numbers.html#numbers.Integral).